crates/db_views/src/report_combined_view.rs

```diff
@@ -167,6 +170,33 @@ impl ReportCombinedQuery {
 
     let conn = &mut get_conn(pool).await?;
 
+    let report_creator_join = post_report::creator_id
+      .eq(report_creator)
+      .or(comment_report::creator_id.eq(report_creator))
+      .or(private_message_report::creator_id.eq(report_creator))
+      .or(community_report::creator_id.eq(report_creator));
+
+    let item_creator_join = post::creator_id
+      .eq(item_creator)
+      .or(comment::creator_id.eq(item_creator))
+      .or(private_message::creator_id.eq(item_creator));
+
+    let resolver_join = private_message_report::resolver_id
+      .eq(resolver)
+      .or(post_report::resolver_id.eq(resolver))
+      .or(comment_report::resolver_id.eq(resolver))
+      .or(community_report::resolver_id.eq(resolver));
+
+    let post_join = post_report::post_id
+      .eq(post::id)
+      .or(comment::post_id.eq(post::id));
+
+    let community_join = community::table.on(
+      community_report::community_id
+        .eq(community::id)
+        .or(post::community_id.eq(community::id)),
+    );
+
```

Comment:
You can move these into a helper function and reuse them for `get_report_count()` too.

Reply:
I'd like to do this for every DB view, but diesel's return types make it near impossible to externalize joins or filters.